src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -703,6 +703,62 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
         billingPostcode: this.giftAidGroup.value.homePostcode,
       });
     }
+
+    if (
+      event.selectedIndex > this.paymentStepIndex &&
```

Comment:
I still have a slight preference for using the step names for new logic, but as we already have this index to maintain it seems reasonable to use it an extra time. I'm less sure about calling the update on both entry to the comms step and entry to the confirm step; seems slightly wasteful while we are not saving comms answers to the Person. Not a blocker.

Reply:
I think we need to do it on both the comms step and the confirm step because it's possible to have already filled in the comms step, go back to change your email address, then skip the comms step and go straight to confirm. We could optimize by checking if the email address has been changed but I'll leave it for now and maybe consider that later.